src/ResourceManager/Compute/Commands.Compute/Extension/Diagnostics/RemoveAzureRmVmssDiagnosticsExtension.cs

```diff
@@ -0,0 +1,93 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Compute.Common;
+using Microsoft.Azure.Management.Compute.Models;
+using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    [Cmdlet(
+        VerbsCommon.Remove,
+        ProfileNouns.VirtualMachineScaleSetDiagnosticsExtension,
+        SupportsShouldProcess = true
+        )]
+    [OutputType(typeof(VirtualMachineScaleSet))]
+    public class RemoveAzureRmVmssDiagnosticsExtension : Microsoft.Azure.Commands.ResourceManager.Common.AzureRMCmdlet
+    {
+        [Parameter(
+            Mandatory = true,
+            Position = 0,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        public VirtualMachineScaleSet VirtualMachineScaleSet { get; set; }
+
+        [Alias("ExtensionName")]
+        [Parameter(
+            Mandatory = false,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The extension name.")]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false,
+            HelpMessage = "To force the removal of the diagnostics extension from the VM scale set.")]
+        public SwitchParameter Force { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            if (ShouldProcess(this.VirtualMachineScaleSet.Name, Properties.Resources.RemoveVmssDiagnosticsExtensionAction))
+            {
+                // VirtualMachineProfile
+                if (this.VirtualMachineScaleSet.VirtualMachineProfile != null &&
+                    this.VirtualMachineScaleSet.VirtualMachineProfile.ExtensionProfile != null &&
+                    this.VirtualMachineScaleSet.VirtualMachineProfile.ExtensionProfile.Extensions != null)
+                {
+                    var extensions = this.VirtualMachineScaleSet.VirtualMachineProfile.ExtensionProfile.Extensions;
+
+                    // Filter by publisher and type
+                    // If extension name is provided, should match the name as well
+                    var diagnosticsExtensions = extensions
+                        .Where(extension =>
+                            DiagnosticsHelper.IsDiagnosticsExtension(extension) &&
+                            (string.IsNullOrEmpty(this.Name) || extension.Name.Equals(this.Name, StringComparison.InvariantCultureIgnoreCase)))
+                        .ToList();
+
+                    if (diagnosticsExtensions.Any())
+                    {
+                        if (Force.IsPresent ||
```

Comment:
Is this really necessary?  Seems liek the ShouldContinue is going to happen every time anythiung is deleted, so not much of a reason for the prompt or for the Force parameter.  I would remove it.